### DISCUSSION TITLE
Value error when uploading mulitple files are optional

### DISCUSSION THREAD

**Discussion by ThinkalVB** [fastapi/fastapi#6226] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/ticket/create/message", tags=["Message component"])
def create_message(ticket_id: int, files: Optional[List[UploadFile]] = File(None), 
    message_text: str=Query(None, max_length=5120), token=Depends(oauth2_scheme)):
    session = Session()
    try:

        session.commit()
        return {"detail":APIResponseE.SUCCESS.name}
    except exc.SQLAlchemyError as e:
        print(str(traceback.format_exception(type(e), e, e.__traceback__)))
        logging.error(str(traceback.format_exception(type(e), e, e.__traceback__)))
        session.rollback()
        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
    finally:
        session.close()

Response
{
  "detail": [
    {
      "loc": [
        "body",
        "files",
        0
      ],
      "msg": "Expected UploadFile, received: <class 'str'>",
      "type": "value_error"
    }
  ]
}
```

### Description

When I am trying to make the uploading multiple files optional I am getting a value error.

### Operating System

Linux

### Operating System Details

Ubuntu

### FastAPI Version

0.78.0

### Python Version

3.8.10

**Comment by JarroVGIT** [comment, marked as answer]
The problem is that you are using two different concepts interchangeable. 

`File()` refers to an uploaded file that is then held in memory. The type of the parameter in your path operator function would be `bytes`. However, `UploadFile` is much more versatile and gives you opportunities to handle multiple larger files without running into OOM errors. When using a parameter in your path operator function with the type `UploadFile()` or `list[UploadFile]`, you don't set the default value to an instance of `File()`. 

I have jotted down a quick overview of this, the following code will run as-is:

```python
from fastapi import FastAPI, File, UploadFile
import uvicorn

app = FastAPI()

@app.get("/")
async def root():
    return {"message":"ok"}

@app.post("/files")
async def files(files: list[UploadFile] | None = None): #The '= None' part makes this an optional parameter!
    return {"message":"ok files"}

@app.post("/file")
async def files(file: bytes | None = File(default=None)): #The '= File(default=None)' part makes this optional
    return {"message":"ok file"}

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8000, ) 
```

Now, using `curl` we can test these endpoints:

```bash
(.venv) jarro@MBP-van-Jarro fastapi-github-issues % curl -X GET http://localhost:8000/
{"message":"ok"}%                                                                                                                     
(.venv) jarro@MBP-van-Jarro fastapi-github-issues % curl -X POST http://localhost:8000/files
{"message":"ok files"}%                                                                                                               
(.venv) jarro@MBP-van-Jarro fastapi-github-issues % curl -X POST http://localhost:8000/file 
{"message":"ok file"}% 
```
As you can see, they return correctly without posting any files with the requests. Hope this clarifies for you!

**Comment by ThinkalVB** [comment]
@JarroVGIT  Yes it do clarifies the issue. Than you for your time.
